packages/text-field/index.tsx

```diff
@@ -320,32 +308,25 @@ class TextField<T extends HTMLElement = HTMLInputElement> extends React.Componen
       textarea,
     } = this.props;
     const {foundation} = this.state;
-    const textField = (
-      <div
-        {...this.otherProps}
-        className={this.classes}
-        onClick={() => foundation && foundation.handleTextFieldInteraction()}
-        onKeyDown={() => foundation && foundation.handleTextFieldInteraction()}
-        key='text-field-container'
-      >
-        {leadingIcon ? this.renderIcon(leadingIcon, onLeadingIconSelect) : null}
-        {foundation ? this.renderInput() : null}
-        {label && !outlined && !fullWidth ? this.renderLabel() : null}
-        {outlined ? this.renderNotchedOutline() : null}
-        {!fullWidth && !textarea && !outlined ? this.renderLineRipple() : null}
-        {trailingIcon ? this.renderIcon(trailingIcon, onTrailingIconSelect) : null}
-      </div>
-    );
 
-    if (helperText) {
-      return (
-        <React.Fragment>
-          {textField}
-          {this.renderHelperText()}
-        </React.Fragment>
-      );
-    }
-    return textField;
+    return (
+      <React.Fragment>
+        <div
+          {...this.otherProps}
+          className={this.classes}
+          onClick={() => foundation!.handleTextFieldInteraction()}
```

Comment:
Looks like you changed this back. Can it not be `onClick={foundation!.handle...}`?

Reply:
Yes we need to use arrow function for "this" keyword.